File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/data/network/model/mapper/AccountMapper.java
package com.lattis.ellipse.data.network.model.mapper;

import androidx.annotation.NonNull;

import com.lattis.ellipse.data.network.base.AbstractDataMapper;
import com.lattis.ellipse.data.network.model.response.UserResponse;
import com.lattis.ellipse.domain.model.Account;

import javax.inject.Inject;

public class AccountMapper extends AbstractDataMapper<UserResponse,Account> {

    @Inject
    public AccountMapper() {}

    @NonNull
    @Override
    public Account mapIn(@NonNull UserResponse userResponse) {
        Account account = new Account(userResponse.getUserId(),null);
        account.setAccessToken(userResponse.getRestToken());
        account.setPassword(userResponse.getRefreshToken());
        account.setUserId(userResponse.getUserId());
        account.setUsersId(userResponse.getUsersId());
        account.setUserType(userResponse.getUserType());
        account.setVerified(userResponse.isVerified());
        return account;
    }

    @NonNull
    @Override
    public UserResponse mapOut(@NonNull Account account) {
        return null;
    }
}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/mapper/AccountMapper.kt
package com.lattis.domain.mapper


import com.lattis.domain.mapper.base.AbstractDataMapper
import com.lattis.domain.models.Account
import com.lattis.domain.models.User
import javax.inject.Inject

class AccountMapper @Inject
constructor() : AbstractDataMapper<User, Account>() {

    override fun mapIn(user: User?): Account {
        val account = Account(user?.id, null)
        account.accessToken = user?.restToken
        account.password = user?.refreshToken
        account.userId = user?.id
        account.usersId = user?.usersId
        account.userType = user?.userType
        account.isVerified = user?.isVerified
        return account
    }

    override fun mapOut(account: Account?): User? {
        return null;
    }
}
